File: src/main/java/net/nodus/database/auth/RefreshTokenRepository.java
package net.nodus.database.auth;

import java.time.Instant;
import java.util.List;
import java.util.Optional;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface RefreshTokenRepository extends MongoRepository<RefreshToken, String> {

    Optional<RefreshToken> findByTokenHashAndRevokedAtIsNullAndDeletedAtIsNull(String tokenHash);

    List<RefreshToken> findByDeletedAtBefore(Instant deletedAt);
}

File: src/main/kotlin/net/nodus/auth/RefreshTokenRepository.kt
package net.nodus.auth

import org.springframework.data.mongodb.repository.MongoRepository
import java.time.Instant

interface RefreshTokenRepository : MongoRepository<RefreshToken, String> {
    fun findByTokenHashAndRevokedAtIsNullAndDeletedAtIsNull(tokenHash: String): RefreshToken?

    fun findByDeletedAtBefore(deletedAt: Instant): List<RefreshToken>
}
